packages/theme-language-server-common/src/json/JSONContributions.ts

```diff
@@ -0,0 +1,120 @@
+import { parseJSON, SourceCodeType } from '@shopify/theme-check-common';
+import {
+  CompletionsCollector,
+  JSONPath,
+  JSONWorkerContribution,
+  MarkedString,
+} from 'vscode-json-languageservice';
+import { AugmentedSourceCode, DocumentManager } from '../documents';
+import { GetTranslationsForURI } from '../translations';
+import { JSONCompletionProvider } from './completions/JSONCompletionProvider';
+import { SchemaTranslationsCompletionProvider } from './completions/providers/SchemaTranslationCompletionProvider';
+import { JSONHoverProvider } from './hover/JSONHoverProvider';
+import { SchemaTranslationHoverProvider } from './hover/providers/SchemaTranslationHoverProvider';
+import { TranslationPathHoverProvider } from './hover/providers/TranslationPathHoverProvider';
+import { RequestContext } from './RequestContext';
+import { findSchemaNode } from './utils';
+
+/** The getInfoContribution API will only fallback if we return undefined synchronously */
+const SKIP_CONTRIBUTION = undefined as any;
+
+/**
+ * I'm not a fan of how json-languageservice does its feature contributions. It's too different
+ * from everything else we do in here.
+ *
+ * Instead, we'll have this little adapter that makes the completions and hover providers feel
+ * a bit more familiar.
+ */
+export class JSONContributions implements JSONWorkerContribution {
+  private hoverProviders: JSONHoverProvider[];
+  private completionProviders: JSONCompletionProvider[];
+
+  constructor(
+    private documentManager: DocumentManager,
+    getDefaultSchemaTranslations: GetTranslationsForURI,
+  ) {
+    this.hoverProviders = [
+      new TranslationPathHoverProvider(),
+      new SchemaTranslationHoverProvider(getDefaultSchemaTranslations),
+    ];
+    this.completionProviders = [
+      new SchemaTranslationsCompletionProvider(getDefaultSchemaTranslations),
+    ];
+  }
+
+  getInfoContribution(uri: string, location: JSONPath): Promise<MarkedString[]> {
+    const doc = this.documentManager.get(uri);
+    if (!doc) return SKIP_CONTRIBUTION;
+    const context = this.getContext(doc);
+    const provider = this.hoverProviders.find((p) => p.canHover(context, location));
+    if (!provider) return SKIP_CONTRIBUTION;
```

Comment:
Because then the contributions that you'd get from the default JSON schema would no longer show up. It's awkward af. 

https://github.com/microsoft/vscode-json-languageservice/blob/main/src/services/jsonHover.ts#L56-L59

I use `SKIP_CONTRIBUTION` because the result is only ignored if you return falsy. And Promise<falsy> is truthy.